### PR TITLE
 ICP-101XX Driver integration

### PR DESCRIPTION
**Pull request by afontaine-invn**
Add ICP-101xx driver sensor folder
Add ICP-101xx in sensor creator module
Also add ICP-101xx CLI support in sensor_test app

### CODE REVIEW

File: hw/drivers/sensors/icp101xx/License.txt [hunk 1]
```diff
@@ -0,0 +1,44 @@
+INVENSENSE, INC. (�LICENSOR�) SOFTWARE LICENSE AGREEMENT (�Agreement�)
+
```
Comment (sterlinghughes) [inline hw/drivers/sensors/icp101xx/License.txt]: Thank you for the PR! 

This license is not an approved license and cannot be included in the repo, for a list of approved licenses please check out: https://apache.org/legal/resolved.html

File: hw/drivers/sensors/icp101xx/syscfg.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    ICP101XX_SHELL_ITF_ADDR:
+        description: 'ICP101XX I2C Address'
+        value: 0x63
+    ICP10114_SHELL_ITF_ADDR:
+        description: 'ICP10114 I2C Address'
+        value: 0x64
+    ICP101XX_ITF_LOCK_TMO:
+        description: 'ICP101XX interface lock timeout in milliseconds'
+        value: 1000 # Max timeout for a full I2C transaction
+    ICP101XX_CLI:
```
Comment (vrahane) [inline hw/drivers/sensors/icp101xx/syscfg.yml]: The CLI should be disabled by default.

File: hw/drivers/sensors/icp101xx/syscfg.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# The BSD License (BSD)
```
Comment (vrahane) [inline hw/drivers/sensors/icp101xx/syscfg.yml]: The Adafruit copyright license should not be used as Sterling suggested.

File: hw/drivers/sensors/icp101xx/src/icp101xx.c [hunk 4]
```diff
@@ -0,0 +1,692 @@
...
+
+/* Exports for the sensor API.*/
+static int icp101xx_sensor_read(struct sensor *, sensor_type_t,
+        sensor_data_func_t, void *, uint32_t);
+static int icp101xx_sensor_get_config(struct sensor *, sensor_type_t,
+        struct sensor_cfg *);
+
+static const struct sensor_driver g_icp101xx_sensor_driver = {
+    icp101xx_sensor_read,
+    icp101xx_sensor_get_config
```
Comment (vrahane) [inline hw/drivers/sensors/icp101xx/src/icp101xx.c]: There is also a set_config function pointer which needs to get set to `config_icp101xx()` here.

File: hw/drivers/sensors/icp101xx/syscfg.yml (hunk 3 above)
Comment (sterlinghughes) [reply hw/drivers/sensors/icp101xx/syscfg.yml]: Adafruit copyright is fine, it's the proprietary license that shouldn't be used.

File: hw/drivers/sensors/icp101xx/syscfg.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+#
+# Copyright (c) 2016 Adafruit Industries
```
Comment (vrahane) [inline hw/drivers/sensors/icp101xx/syscfg.yml]: The Adafruit copyright license should not be used as Sterling suggested.

File: hw/sensor/creator/pkg.yml [hunk 6]
```diff
@@ -62,5 +62,9 @@ pkg.deps.KXTJ3_OFB:
     - "@apache-mynewt-core/hw/drivers/sensors/kxtj3"
 pkg.deps.DPS368_OFB:
     - "@apache-mynewt-core/hw/drivers/sensors/dps368"
+pkg.deps.ICP101XX_OFB:
+    - "@apache-mynewt-core/hw/drivers/sensors/icp101xx"
+pkg.deps.ICP10114_OFB:
+    - "@apache-mynewt-core/hw/drivers/sensors/icp101xx"
```
Comment (vrahane) [inline hw/sensor/creator/pkg.yml]: Shouldn't the dependency be @apache-mynewt-core/hw/drivers/sensors/icp10114 ?

File: hw/drivers/sensors/icp101xx/src/icp101xx.c [hunk 7]
```diff
@@ -0,0 +1,692 @@
...
+    STATS_SECT_ENTRY(errors)
+STATS_SECT_END
+
+/* Define stat names for querying */
+STATS_NAME_START(icp101xx_stat_section)
+    STATS_NAME(icp101xx_stat_section, errors)
+STATS_NAME_END(icp101xx_stat_section)
+
+/* Global variable used to hold stats data */
+STATS_SECT_DECL(icp101xx_stat_section) g_icp101xxstats;
```
Comment (vrahane) [inline hw/drivers/sensors/icp101xx/src/icp101xx.c]: I know you were testing multiple instances and seeing issues with statistics, in my latest commit to mynewt-core: https://github.com/apache/mynewt-core/commit/6acbbe728e013aa6f786f8721be29af888de9a81#diff-6dcb21895ea4bbd985ecdff5e0f9372eR281, I have fixed it for the dps368 sensor. You could fix your PR accordingly.

File: hw/drivers/sensors/icp101xx/syscfg.yml (hunk 3 above)
Comment (afontaine-invn) [reply hw/drivers/sensors/icp101xx/syscfg.yml]: The Adafruit license was copied from another driver that we have as example. Could we put instead the Invensence license ?

Comment (sterlinghughes) [reply hw/drivers/sensors/icp101xx/syscfg.yml]: the invense license is the problem!  the adafruit license is fine.  I posted a link above for acceptable third party licenses -- 3-clause BSD is OK, however Invensense's proprietary license is not.

File: hw/drivers/sensors/icp101xx/License.txt (hunk 1 above)
Comment (afontaine-invn) [reply hw/drivers/sensors/icp101xx/License.txt]: Licence updated for a BSD 3-Clause license

File: hw/drivers/sensors/icp101xx/syscfg.yml (hunk 2 above)
Comment (afontaine-invn) [reply hw/drivers/sensors/icp101xx/syscfg.yml]: Thanks, the ICP101XX_CLI is now disabled

File: hw/drivers/sensors/icp101xx/src/icp101xx.c (hunk 4 above)
Comment (afontaine-invn) [reply hw/drivers/sensors/icp101xx/src/icp101xx.c]: The icp101xx_sensor_set_config() has been added

File: hw/drivers/sensors/icp101xx/src/icp101xx.c (hunk 7 above)
Comment (afontaine-invn) [reply hw/drivers/sensors/icp101xx/src/icp101xx.c]: Ok, thanks. I updated the driver to handle multiple stats instance as you suggest.

**Pull request merged**